Swift 6: Potential race warnings in NetworkTransport continuation handlers

**Describe the bug**
Compilation failure with **Swift 6.3** on **Xcode 26.4 beta 2 (17E5170d)**

**To Reproduce**
1.  `swift build`

**Expected behavior**: `Build complete!`


**Logs**

```
../swift-sdk/Sources/MCP/Base/Transports/NetworkTransport.swift:532:33: error: sending 'sendContinuationResumed' risks causing data races [#SendingRisksDataRace]
530 |
531 |                         Task { @MainActor in
532 |                             if !sendContinuationResumed {
    |                                 |- error: sending 'sendContinuationResumed' risks causing data races [#SendingRisksDataRace]
    |                                 `- note: task-isolated 'sendContinuationResumed' is captured by a main actor-isolated closure. main actor-isolated uses in closure may race against later nonisolated uses
533 |                                 sendContinuationResumed = true
534 |                                 if let error = error {

../swift-sdk/Sources/MCP/Base/Transports/NetworkTransport.swift:763:29: error: sending 'receiveContinuationResumed' risks causing data races [#SendingRisksDataRace]
761 |                     content, _, isComplete, error in
762 |                     Task { @MainActor in
763 |                         if !receiveContinuationResumed {
    |                             |- error: sending 'receiveContinuationResumed' risks causing data races [#SendingRisksDataRace]
    |                             `- note: task-isolated 'receiveContinuationResumed' is captured by a main actor-isolated closure. main actor-isolated uses in closure may race against later nonisolated uses
764 |                             receiveContinuationResumed = true
765 |                             if let error = error {

[#SendingRisksDataRace]: <https://docs.swift.org/compiler/documentation/diagnostics/sending-risks-data-race>
````

**Additional context**

```
# swift --version
swift-driver version: 1.148.6 Apple Swift version 6.3 (swiftlang-6.3.0.121.2 clang-2100.0.121.2)
Target: arm64-apple-macosx26.0
```

**Version 26.4 beta 2 (17E5170d)**

--- a/Sources/MCP/Base/Transports/NetworkTransport.swift
+++ b/Sources/MCP/Base/Transports/NetworkTransport.swift
@@ -511,8 +511,8 @@ import Logging
             var messageWithNewline = message
             messageWithNewline.append(UInt8(ascii: "\n"))
 
-            // Use a local actor-isolated variable to track continuation state
-            var sendContinuationResumed = false
+            // Use a main-isolated variable to track continuation state
+            let sendContinuationResumed = MainFlag()
 
             try await withCheckedThrowingContinuation {
                 [weak self] (continuation: CheckedContinuation<Void, Swift.Error>) in
@@ -529,8 +529,8 @@ import Logging
                         guard let self = self else { return }
 
                         Task { @MainActor in
-                            if !sendContinuationResumed {
-                                sendContinuationResumed = true
+                            if !sendContinuationResumed.flag {
+                                sendContinuationResumed.flag = true
                                 if let error = error {
                                     self.logger.error("Send error: \(error)")
 
@@ -747,7 +747,8 @@ import Logging
         /// - Returns: The received data chunk
         /// - Throws: Network errors or transport failures
         private func receiveData() async throws -> Data {
-            var receiveContinuationResumed = false
+            // Use a main-isolated variable to track continuation state
+            let receiveContinuationResumed = MainFlag()
 
             return try await withCheckedThrowingContinuation {
                 [weak self] (continuation: CheckedContinuation<Data, Swift.Error>) in
@@ -760,8 +761,8 @@ import Logging
                 connection.receive(minimumIncompleteLength: 1, maximumLength: maxLength) {
                     content, _, isComplete, error in
                     Task { @MainActor in
-                        if !receiveContinuationResumed {
-                            receiveContinuationResumed = true
+                        if !receiveContinuationResumed.flag {
+                            receiveContinuationResumed.flag = true
                             if let error = error {
                                 continuation.resume(throwing: MCPError.transportError(error))
                             } else if let content = content {
@@ -787,6 +788,7 @@ import Logging
             self.isConnected = connected
         }
     }
+    @MainActor final class MainFlag { var flag = false }
 
     extension NWError {
         /// Whether this error indicates a connection has been lost or reset.

